flutter/lib/src/native/java/android_replay_recorder.dart

```diff
@@ -48,3 +73,141 @@ class AndroidReplayRecorder extends ScheduledScreenshotRecorder {
     }
   }
 }
+
+// Based on https://dart.dev/language/isolates#robust-ports-example
+class _AndroidNativeReplayWorker {
+  final SendPort _commands;
+  final ReceivePort _responses;
+  final Map<int, Completer<Object?>> _activeRequests = {};
+  int _idCounter = 0;
+  bool _closed = false;
+
+  static Future<_AndroidNativeReplayWorker> spawn() async {
+    // Create a receive port and add its initial message handler
+    final initPort = RawReceivePort();
+    final connection = Completer<(ReceivePort, SendPort)>.sync();
+    initPort.handler = (SendPort commandPort) {
+      connection.complete((
+        ReceivePort.fromRawReceivePort(initPort),
+        commandPort,
+      ));
+    };
+
+    // Spawn the isolate.
+    try {
+      await Isolate.spawn(_startRemoteIsolate, (initPort.sendPort),
+          debugName: 'SentryReplayRecorder');
+    } on Object {
+      initPort.close();
+      rethrow;
+    }
+
+    final (ReceivePort receivePort, SendPort sendPort) =
+        await connection.future;
+
+    return _AndroidNativeReplayWorker._(receivePort, sendPort);
+  }
+
+  _AndroidNativeReplayWorker._(this._responses, this._commands) {
+    _responses.listen(_handleResponsesFromIsolate);
+  }
+
+  Future<Object?> nativeAddScreenshot(_WorkItem item) async {
+    if (_closed) throw StateError('Closed');
+    final completer = Completer<Object?>.sync();
+    final id = _idCounter++;
+    _activeRequests[id] = completer;
+    _commands.send((id, item));
+    return await completer.future;
+  }
+
+  void _handleResponsesFromIsolate(dynamic message) {
+    final (int id, Object? response) = message as (int, Object?);
+    final completer = _activeRequests.remove(id)!;
+
+    if (response is RemoteError) {
+      completer.completeError(response);
+    } else {
+      completer.complete(response);
+    }
+
+    if (_closed && _activeRequests.isEmpty) _responses.close();
+  }
+
+  /// This is the actual Android native implementation, the rest is just plumbing.
+  static void _handleCommandsToIsolate(
+    ReceivePort receivePort,
+    SendPort sendPort,
+  ) {
+    // Android Bitmap creation is a bit costly so we reuse it between captures.
+    native.Bitmap? bitmap;
+
+    final _nativeReplay = native.SentryFlutterPlugin$Companion(null)
```

Comment:
Please open an issue for JNIgen to remove the synthetic "default constructor" marker added by the Kotlin compiler from the generated code and just pass `null`.